addons/resource/populate/resource_calendar.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import random
+
+from odoo import models
+from odoo.tools import populate
+
+
+class ResourceCalendar(models.Model):
+    _inherit = "resource.calendar"
+
+    _populate_dependencies = ["res.company"]  # multi-company setup
+
+    _populate_sizes = {
+        "small":  10,  # 1-2 per company
+        "medium": 30,  # 3 per company
+        "large":  250  # 5 per company
+    }
+
+    def _populate_factories(self):
+        company_ids = self.env.registry.populated_models["res.company"]
+
+        return [
+            ("company_id", populate.iterate(company_ids)),
+            ("name", populate.iterate(
+                ["A little {counter}", "A lot {counter}"])),
+        ]
+
+    def _populate(self, size):
+        records = super()._populate(size)
+
+        # Randomly remove 1 half day from schedule
+        a_lot = self.search([("name", "like", "A lot")])
+        for record in a_lot:
+            to_pop = random.randint(0, 9)
+            record.attendance_ids = (
+                self.env["resource.calendar.attendance"].browse([
+                    attendance_id
+                    for (idx, attendance_id)
+                    in enumerate(record.attendance_ids.ids)
+                    if idx != to_pop
+                ])
+            )
```

Comment:
You could do something like this also, to remove the `to_pop`th attendance:
```suggestion
            att_id = record.attendance_ids[to_pop].id
            record.write({
                'attendance_ids': [(3, att_id, 0)],
            })
```